components/Image.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+/**
+ * Part of GDL gdl-frontend.
+ * Copyright (C) 2017 GDL
+ *
+ * See LICENSE
+ */
+import * as React from 'react';
+import styled from 'styled-components';
+import { width } from 'styled-system';
+import height from './helpers/height';
+
+// Base on gatbsy-image Kyle Mathews
+// See https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-image/src/index.js
+
+// Cache if we've seen an image before so we don't bother with lazy loading and fading on subsequent mounts
+const imageCache = {};
+
+function inImageCache(props): boolean {
+  if (imageCache[props.src]) {
+    return true;
+  }
+  imageCache[props.src] = true;
```

Comment:
Is this the best place to set this? `inImageCache` looks like a function that shouldn't have any side effects, and that this assignment should be done somewhere else.